src/dist/http.rs

```diff
@@ -1078,7 +1078,7 @@ mod client {
     use super::urls;
     use crate::errors::*;
 
-    const REQUEST_TIMEOUT_SECS: u64 = 600;
+    const REQUEST_TIMEOUT_SECS: u64 = 1200;
```

Comment:
5 minutes is too short to compile some complex kernels, so I bumped to 10 minutes here. It would be good to make this configurable.